Class/Practise 10.1/src/TeamSorter.cpp

```diff
@@ -0,0 +1,25 @@
+#include "TeamSorter.hpp"
+#include <algorithm>
+#include <random>
+
+TeamSorter::TeamSorter() {
+    criteria.push_back([](const Team& a, const Team& b) { return a.getTotalPoints() > b.getTotalPoints(); });
+    criteria.push_back([](const Team& a, const Team& b) { return a.getTotalTopHeight() > b.getTotalTopHeight(); });
+    criteria.push_back([](const Team& a, const Team& b) { return a.getGoalDifference() > b.getGoalDifference(); });
+    criteria.push_back([](const Team& a, const Team& b) { return a.getTotalGoalsScored() > b.getTotalGoalsScored(); });
+    criteria.push_back([](const Team& a, const Team& b) { return a.getTotalRedCards() < b.getTotalRedCards(); });
+    criteria.push_back([](const Team& a, const Team& b) { return a.getTotalYellowCards() < b.getTotalYellowCards(); });
+}
+
+void TeamSorter::sortTeams(std::vector<Team>& teams) const {
+    auto compare = [this](const Team& a, const Team& b) { return compareTeams(a, b); };
+    std::sort(teams.begin(), teams.end(), compare);
+}
+
+bool TeamSorter::compareTeams(const Team& a, const Team& b) const {
+    for (const auto& criterion : criteria) {
+        if (criterion(a, b)) return true;
```

Comment:
those 2 lines looks strange. Also I am not sure that we need that list of lambda. For comparison we can use <= to have fully ordered relation
Or we can avoid creation of this class just by adding spaceship operator<=> to Team class. Then std::sort can work directly with Team objects